app/services/creators/capital_eligibility_creator.rb

```diff
@@ -25,6 +25,20 @@ def create_eligibility(ptc)
           lower_threshold: controlled_threshold(ptc),
           assessment_result: "pending",
         )
+      elsif ptc.to_sym == CFEConstants::IMMIGRATION_PROCEEDING_TYPE_CCMS_CODE
```

Comment:
I think we're making this class a lot more complicated than we need to. We've got a 4-block if/else statement here and another if block inside `controlled_threshold`. The problem is that we're splitting up the logic by level of help first and proceeding type second, but actually the proceeding type affects both controlled and certificated work in the same way. And there's only one special case, which is Immigration. So I think you can safely drop the whole `ptc.to_sym == CFEConstants::ASYLUM_PROCEEDING_TYPE_CCMS_CODE` branch, and if you pull out the `ptc.to_sym == CFEConstants::IMMIGRATION_PROCEEDING_TYPE_CCMS_CODE` above the check for controlled/certificated you can get that out of the way in one fell swoop, and make this whole class much less complex.

Reply:
Although I've missed something in the above: lower limits not applying in for immigration/asylum proceeding types for certificated cases. So maybe it's not quite as neat as I thought.

Reply:
Yes I've been grappling with the clearest way to code what is essentially a very simple rule!